### ISSUE TITLE
How to add API token without credentials 

### ISSUE THREAD

**Issue by luis-alves** [MCMicS/jenkins-control-plugin#262]
Hi,
Jenkins is hosted in my company VPN, I don't need credentials to have access to the pipelines, so I don't have any to login. To get an API Token to place on the plugin setup, I need to login, but since I don't have the credentials I'm not able to do it. How can I solve this issue.
Thanks

**Comment by MCMicS** [comment]
I can have take a look in new year. 
You have no authentification in jenkins server configuration right?

**Comment by luis-alves** [comment]
Yes, that is correct. I have no way to login my self into Jenkins or access configure path.

**Comment by MCMicS** [comment]
I can try it it in new year because im on holiday at this time. But i will look to it to use without given credentials

**Comment by luis-alves** [comment]
I managed to make the plugin retrieve the data by simply replacing the post method by a simple get method. Unfortunately "Build on Jenkins" command does not work because it requires a post method and then it returns an error related with the Crumb data.

**Comment by MCMicS** [comment]
Sorry for delay. I will look at this as soon as possible

**Comment by MCMicS** [comment]
Sorry for the delay:
 Which Jenkins Version to you use?

Because Jenkins says: 
> Beginning with Jenkins 2.214 and Jenkins LTS 2.222.1, the "Enable Security" checkbox has been removed. Jenkins own user database is used as the default security realm.

So you also know which security settings is used?
![grafik](https://user-images.githubusercontent.com/4604206/104953041-f04dec00-59c5-11eb-9920-1c035f13a496.png)

**Comment by MCMicS** [comment]
@luis-alves It is not possible to use an authentification provider within jenkins?
For security reason it is bette to use authetificated users.
API token are currently available for users. There is an open feature to support Token for Jobs (https://issues.jenkins.io/browse/JENKINS-56465)

But for now I suggest to create user in jenkins and use the API Token instead because else you provide a crumb data which is changed often and not recommend more for API calls.

**Comment by luis-alves** [comment]
@MCMicS I can't login because I don't have an account. I access Jenkins through a VPN, so the authentication is made in it, and Jenkins is totally opened. The version is Jenkins 2.263.1, and I don't have admin privileges to tell you which settings are currently set.

**Comment by MCMicS** [comment]
Hi @luis-alves 
Can you provide a screenshot from you Jenkins landing page?
Is there a logged-in user after you redirected from you VPN and proxy to jenkins?

It is possible to grant me temporary access to the server?

**Comment by MCMicS** [comment]
Hi @luis-alves 
Jenkins recommend to use Token based for script (see https://www.jenkins.io/doc/book/security/csrf-protection/).

I've not found any solution except following:
1. Allow only GET calls to list project. But then some features are missing (disabled) like Build and so on
2. Ask for Crumb Issuer for anonymous access and use this
3. Usage of ' https://github.com/jenkinsci/build-token-root-plugin'  on Jenkins server and try to build with the created access token

In my opionion the secure way is the first. A dedicaterd userr for script access. If this is still not possible then try out the crumb way

Is the given setup still valid in your environment?

**Comment by doukhahmed** [comment]
Hello

How to configure the Jenkins plugin in IntelliJ IDEA to connect with a **local** Jenkins server?

what are the params for : 
- Servers address 
- Jenkins URL 
- Username 
- API-Token

**Comment by MCMicS** [comment]
@doukhahmed Please do not post same questions twice; Answer found at: https://github.com/MCMicS/jenkins-control-plugin/discussions/529

**Comment by doukhahmed** [comment]
@MCMicS, sorry and thanks for the answer! :)
